orca-pipelinetemplate/src/main/java/com/netflix/spinnaker/orca/pipelinetemplate/v1schema/model/StageDefinition.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.orca.pipelinetemplate.v1schema.model;
+
+import java.util.List;
+import java.util.Map;
+
+public class StageDefinition implements NamedContent {
+
+  private String id;
+  private InjectionRule inject;
+  private String dependsOn; // TODO rz - comma-delimited for now
+  private String type;
+  private Map<String, Object> config;
+  private List<Map<String, Object>> notifications;
+  private String comments;
+
+  @Override
+  public String getName() {
+    return id;
+  }
+
+  public static class InjectionRule {
```

Comment:
Yep. In my mind, inject is a "break glass in case of fire" or "mostly the same" feature. Let's say the company paved road policy fits your use case almost exactly and you want to use it, but your team is responsible for a new canary feature that should be run just before the deploy stage. You want to stay in-sync with the paved road, but also want to offer teams the "canary-enabled paved road".

```yaml
# Paved road pipeline template
# spinnaker://netflix/pavedRoad
id: pavedRoad
stages:
- id: findImage
  type: findImageFromTags
  config: '{ /* */ }'
- id: deploy
  type: deploy
  config: '{ /* */ }
```
  
```yaml
# canary team + paved road pipeline template
# spinnaker://canaryteam/pavedRoad
id: canaryPavedRoad
source: spinnaker://pavedRoad
variables:
- name: canaryScoreThreshold
  description: The score (0-100) which a canary should meet to be considered a success.
  type: int
stages:
- id: canary
  inject:
    before: findImage
  type: canary
  config:
    scoreThreshold: "{{ canaryScoreThreshold }}"
```
  
Then, the canary team starts distributing their version of the pipeline for teams, so a team switches their pipeline to use the new canary-enhanced pipeline. Once canary becomes best-practice, it'd get rolled into upstream and then eventually the canary team would deprecate their pipeline (somehow - not part of this scope).

```diff
# spinnaker://myTeam/myApp
id: myApp
pipeline:
  template:
-    source: spinnaker://netflix/pavedRoad
+    source: spinnaker://canaryteam/pavedRoad
  variables:
    canaryScoreThreshold: 90
    # ...
```